deepchem/molnet/load_function/zinc15_datasets.py

```diff
@@ -0,0 +1,233 @@
+"""
+ZINC15 commercially-available compounds for virtual screening.
+"""
+import os
+import logging
+import deepchem
+from deepchem.feat import Featurizer
+from deepchem.trans import Transformer
+from deepchem.splits.splitters import Splitter
+from deepchem.molnet.defaults import get_defaults
+
+from typing import List, Tuple, Dict, Optional
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_DIR = deepchem.utils.data_utils.get_data_dir()
+
+# dict of accepted featurizers for this dataset
+DEFAULT_FEATURIZERS = get_defaults("feat")
+
+# Names of supported featurizers
+zinc15_featurizers = [
+    'SmilesToImage', 'OneHotFeaturizer', 'SmilesToSeq', 'RDKitDescriptors',
+    'ConvMolFeaturizer', 'WeaveFeaturizer', 'CircularFingerprint',
+    'Mol2VecFingerprint'
+]
+DEFAULT_FEATURIZERS = {k: DEFAULT_FEATURIZERS[k] for k in zinc15_featurizers}
+
+# dict of accepted transformers
+DEFAULT_TRANSFORMERS = get_defaults("trans")
+
+# dict of accepted splitters
+DEFAULT_SPLITTERS = get_defaults("splits")
+
+# names of supported splitters
+zinc15_splitters = ['RandomSplitter', 'RandomStratifiedSplitter']
+DEFAULT_SPLITTERS = {k: DEFAULT_SPLITTERS[k] for k in zinc15_splitters}
+
+
+def load_zinc15(
+    featurizer=DEFAULT_FEATURIZERS['OneHotFeaturizer'],
+    transformers: List = [DEFAULT_TRANSFORMERS['NormalizationTransformer']],
+    splitter=DEFAULT_SPLITTERS['RandomSplitter'],
+    reload: bool = True,
+    data_dir: Optional[str] = None,
+    save_dir: Optional[str] = None,
+    featurizer_kwargs: Dict[str, object] = {},
+    splitter_kwargs: Dict[str, object] = {},
+    transformer_kwargs: Dict[str, Dict[str, object]] = {
+        'NormalizationTransformer': {
+            'transform_X': True
+        }
+    },
+    zinc15_kwargs: Dict[str, str] = {
+        'dataset_size': '250K',
+        'dataset_dimension': '2D'
+    },
+    **kwargs) -> Tuple[List, Optional[Tuple], List]:
+  """Load zinc15.
+
+  ZINC15 is a dataset of over 230 million purchasable compounds for
+  virtual screening of small molecules to identify structures that
+  are likely to bind to drug targets. ZINC15 data is currently available
+  in 2D (SMILES string) format.
+
+  MolNet provides subsets of 250K, 1M, and 10M "lead-like" compounds
```

Comment:
Could we also add the full 235M compound dataset as an option? That might be useful for large pretraining applications

Reply:
Certainly, that would be cool to have access to. It may take me a bit to wrangle that into a CSV but I'll work on it.